### DISCUSSION TITLE
Language Navigation

### DISCUSSION THREAD

**Discussion by Anton-Olivier** [nl-design-system/backlog#245] [Component suggestions]
<img width="480" alt="visuele weergave van het Language navigation component" src="https://github.com/nl-design-system/backlog/assets/545207/474e3607-2bfd-4c85-b8eb-566f0d946762">

We willen graag meer leren over het Language Navigation component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1714125934150889)

---

## Naam
Language Navigation

## Doel
Eén of meerdere links om de website te openen in een andere taal.

## Ook bekend als
Language Switcher, Language Selection

## Onderzoek
[Overzicht van Language Switchers op websites](https://docs.google.com/spreadsheets/d/1g-o0qbx5oJ9jngB4WWAG4g7fpIDfxSuGK59gKvoI7uU/edit?usp=sharing/)
Spreadsheet van sites met taalkeuze componenten: overheid, e-commerce, luchtvaartmaatschappijen en brandidentiteit uit binnen- en buitenland. Er kan gesorteerd en gefilterd worden op een aantal kenmerken als type en aantal talen. Graag aanvullen als je extra voorbeelden hebt.

[6 Tips for Improving Language Switchers on Ecommerce Sites - Nielsen Norman Group](https://www.nngroup.com/articles/language-switching-ecommerce/)
Artikel met tips voor het Language Switcher component op e-commerce sites. De meeste tips zijn ook bruikbaar voor overheidssites. Deze informatie komt ook terug in een [bijbehorende video](https://www.nngroup.com/videos/language-switchers/).

[Designing A Perfect Language Selector UX - Smashing magazine](https://www.smashingmagazine.com/2022/05/designing-better-language-selector/)
Artikel waarin veel verschillende voorbeelden voorbij komen. Er wordt ook geschreven over het gebruik van re-directs, vlaggen, iconen en de schrijfwijze van labels.

[Supporting multiple languages - Github Discussion GOV.UK](https://github.com/alphagov/govuk-design-system-backlog/issues/99)
Github Discussion van GOV.UK rondom het kunnen selecteren van verschillende talen. In de comments komt ook een verwijzing naar het component '[Translate navigation](https://components.publishing.service.gov.uk/component-guide/translation_nav)' voorbij.

[Aria-label is a xenophobe](https://heydonworks.com/article/aria-label-is-a-xenophobe/)
Artikel waarin wordt uitgelegd dat teksten die code binnen een 'aria-label' staan niet goed worden vertaald.

[WCAG Success Criterion 3.1.2 - Language of Parts](https://www.w3.org/WAI/WCAG21/Understanding/language-of-parts.html#:~:text=Alternative%20language%20links)
In dit Success Criterion wordt een voorbeeld gegeven van code die bij dit component kan helpen.

[translate="no"](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/translate)
Stukje code dat ervoor zorgt dat bepaalde teksten juist niét vertaald worden. Denk aan de achternamen (Bakker of Groot), straatnamen, etc.

[Figma file - Logius: Language Navigation](https://www.figma.com/design/BJ17S28XNFopx4VZ0jFMy2/Taalswitch?node-id=1-1679&t=fJ8ycbu7u3QFxGpH-1)
In dit projectbestand staat verzameld onderzoek met visuele voorbeelden en uittreksels van internet artikelen.

[Onderzoek 'Taalswitch' door Logius - Youtube](https://youtu.be/WgA8flJjCZ0?feature=shared)
Sabine vertelt over de taalswitch, ook wel bekend als de Language Navigation component. Ze deelt inzichten uit het gebruikersonderzoek dat hiernaar is gedaan.

**Comment by Anton-Olivier** [comment]
[Google Translate gebruik op utrecht.nl | Gebruikersonderzoeken](https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/website-gebruik/utrecht-google-translate)

[Google Translate gebruik op utrecht.nl | Gebruikte code](https://robbert.github.io/translate-test/)

**Comment by Anton-Olivier** [comment]
[Guidelines van de Nieuw-Zeelandse regering over hoe om te gaan met weergeven van andere talen](https://www.digital.govt.nz/standards-and-guidance/design-and-ux/content-design-guidance/inclusive-language/providing-information-in-other-languages/)

**Comment by JeroenduC** [comment]
Interessant stuk over de toepasbaarheid van een language switcher, voegt deze ook echt waarde toe? Als de content van de site goed en duidelijk is geschreven (B1, duidelijke korte zinnen, geen jargon), kan een automatische vertaler de site goed vertalen. We zien mensen met een andere taalachtergrond, die Nederlands net nog wat ingewikkeld vinden, standaard Google Translate gebruiken om stukken of hele sites te vertalen. Ze gaan niet op zoek naar een language switcher van de site zelf.

**Reply by JeroenduC** [reply]
Misschien is de vraag meer: In welke situatie heeft een taal switcher zin? Wanneer is het schrijven  van goede, vertaalbare content waardevoller dan het inzetten van een taal switcher?

**Reply by Anton-Olivier** [reply]
@JeroenduC : Wat je schrijft lijkt uitleg te zijn voor een link naar een artikel... Begrijp ik wat je schrijft verkeerd of zou je een link toe kunnen voegen?

**Reply by JeroenduC** [reply]
Excuus voor de verwarring, er is geen artikel, meer dat ik het onderwerp interessant vind :).

**Comment by Anton-Olivier** [comment]
[Artikel: Hoe bied je je website het beste meertalig aan? - OpenGemeenten](https://www.linkedin.com/pulse/hoe-bied-je-website-het-beste-meertalig-aan-opengemeenten-trjme/?trackingId=%2BfqqGOfnmre%2FP4A8PWyJ4g%3D%3D)

Met dank aan Martijn Rietveld van OpenGemeenten

**Comment by MariekeBrouwer** [comment]
gemeente Groningen (en denk gemeenten vanuit het Open Online dimpact-verband) gaan/ kunnen gebruik maken van de DeepL plugin op hopelijk kortetermijn. We hebben daarin het voorbeeld van Rotterdam.nl gevolgd (maar laten de verwijzing naar toeristische website achterwege). 
![Homepage_translate_expanded (1)](https://github.com/nl-design-system/backlog/assets/149144374/71aac614-c4fc-4227-aa71-c3a3243e8785)


![Hompage_Menu_expanded (1)](https://github.com/nl-design-system/backlog/assets/149144374/ab3226cf-a658-4ff1-900d-57ba9cfbac5f)


Op desktop is het een dropdown voor selectie, op mobiel een language selector met radio button met een backdrop. Zijn vrij snelle / slordige opzetjes voor mobiel trouwens voor ons ter voorbeeld , dus de spacing klopt daar nog even niet helemaal :)

**Comment by jeffreylauwers** [comment]
## Amsterdam

Geen specifiek component gevonden in Figma of Storybook. Wel een mogelijkheid om van taal te wisselen op https://www.amsterdam.nl/

---

### Screenshots / Varianten

**Website**

Small viewport
![Scherm­afbeelding 2024-05-01 om 14 09 35](https://github.com/nl-design-system/backlog/assets/545207/cc51daf7-28ee-44a8-b122-57985f2270be)

Small viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 14 00 40](https://github.com/nl-design-system/backlog/assets/545207/d0f60181-22da-41b8-9bd0-c7fd9dfe0232)

Small viewport - Engels
![Scherm­afbeelding 2024-05-01 om 14 00 49](https://github.com/nl-design-system/backlog/assets/545207/375cb289-82a7-4dc3-872a-eb5df42cc1f7)

Large viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 14 00 17](https://github.com/nl-design-system/backlog/assets/545207/ca4b47a4-ab38-48aa-b506-42f593128e44)

Large viewport - Engels
![Scherm­afbeelding 2024-05-01 om 14 00 26](https://github.com/nl-design-system/backlog/assets/545207/9f5dda17-4eed-47e2-b7ac-90303d86230a)

---

### Notities

- Twee talen: Engels en Nederlands
- URL wijzigt bij keuze voor engels naar: https://www.amsterdam.nl/en/
- Bij small viewport als laatste item van 'hamburger' menu.
- Bij large viewport als link (label: 'English site' icon: chevron-right) in header (positie: inline-end).
- Label wisselt van 'English site' naar 'Dutch site'.

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=24631-126297&mode=design&t=wfNFW7NXmxPDTL33-4)
- Mogelijkheid om van taal te wisselen op https://www.denhaag.nl

---

### Screenshots / Varianten

**Figma**

Small viewport
![Scherm­afbeelding 2024-05-01 om 16 10 45](https://github.com/nl-design-system/backlog/assets/545207/3d6f76da-03ad-4365-a426-53734806c30b)

Large viewport
![Scherm­afbeelding 2024-05-01 om 16 10 52](https://github.com/nl-design-system/backlog/assets/545207/2b6402c1-a314-466f-a92e-e9b517a1cc85)



**Website**

Small viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 14 10 36](https://github.com/nl-design-system/backlog/assets/545207/bea470f0-02e0-4cfa-a8d8-47b96e7ffc49)

Small viewport - Nederlands (menu)
![Scherm­afbeelding 2024-05-01 om 14 08 03](https://github.com/nl-design-system/backlog/assets/545207/cd184c8c-f1e4-494b-84cc-d867e814a9a7)

Small viewport - Engels (menu)
![Scherm­afbeelding 2024-05-01 om 14 08 18](https://github.com/nl-design-system/backlog/assets/545207/ae036658-33ff-45b2-bc5c-18459c1abf4e)

Large viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 14 08 47](https://github.com/nl-design-system/backlog/assets/545207/5c2c2ea7-03a5-4b40-a2e1-3879dedd22b1)

Large viewport - Nederlands (translate active)
![Scherm­afbeelding 2024-05-01 om 14 08 54](https://github.com/nl-design-system/backlog/assets/545207/30cd2316-09f3-4f78-b582-201d802845f2)

Large viewport - Engels
![Scherm­afbeelding 2024-05-01 om 14 09 02](https://github.com/nl-design-system/backlog/assets/545207/65a8e872-3523-49ee-99cc-cad283b31b77)

Large viewport - Engels (translate active)
![Scherm­afbeelding 2024-05-01 om 14 09 09](https://github.com/nl-design-system/backlog/assets/545207/685d37cb-79ea-46af-b3be-96c38ec54671)

---

### Notities

Figma
Keuze tussen link naar engelse versie en vertalen via Google Translate door middel van Select component.

Website
- Twee talen: Engels en Nederlands
- URL denhaag.nl gaat direct naar: https://www.denhaag.nl/nl/
- URL wijzigt bij keuze naar: https://www.denhaag.nl/en/
- Small viewport als laatste item van 'hamburger' menu.
- Bij large viewport triggert button (label: 'NL' + icon: chevron-down) in header (positie: inline-end) een overlay. Hierin staat de link.
- Genaamd: Language switch
- Label van button wisselt van 'NL' naar 'EN'.

**Comment by jeffreylauwers** [comment]
## Open formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/docs/composites-language-selection--docs)

---

### Screenshots / Varianten

**Storybook**

![Scherm­afbeelding 2024-05-01 om 14 36 55](https://github.com/nl-design-system/backlog/assets/545207/c6bdceb8-4e9f-4ade-8881-d422e94ea517)

---

### Notities

- Genaamd: Language selection
- Ook een [functioneel component](https://open-formulieren.github.io/open-forms-sdk/?path=/docs/composites-language-selection--docs#functional) beschikbaar dat het 'slimme' gedrag afhandeld.

**Comment by jeffreylauwers** [comment]
## OpenGemeenten

Geen specifiek component gevonden in Figma of Storybook. Zoals toegelicht door Martijn Rietveld (OpenGemeenten) tijdens een Design Open Dag is er wel een mogelijkheid om van taal te wisselen op https://www.nieuwegein.nl/ via Google Translate.

---

### Screenshots / Varianten

**Website**

Small viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 14 48 39](https://github.com/nl-design-system/backlog/assets/545207/a70d7655-d5f1-4cd5-a125-f0d6bb1f8f80)

Large viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 14 45 27](https://github.com/nl-design-system/backlog/assets/545207/9fd4a781-1648-4459-866f-a004288b60b1)

Large viewport - Nederlands (pagina)
![Scherm­afbeelding 2024-05-01 om 14 45 42](https://github.com/nl-design-system/backlog/assets/545207/c55fbcb5-9cd4-4c50-b3d4-dad5f6b5d6ed)

Large viewport - Engels (Google Translate - bar)
![Scherm­afbeelding 2024-05-01 om 14 46 04](https://github.com/nl-design-system/backlog/assets/545207/dabb8c7b-af1f-4f5e-9de5-fcab0ea022c2)

Large viewport - Engels (Google Translate - sticky button)
![Scherm­afbeelding 2024-05-01 om 14 46 17](https://github.com/nl-design-system/backlog/assets/545207/c7dab9df-48da-4f1a-a77e-98fce9dde3d2)

---

### Notities

- URL wijzigt bij keuze naar: https://www-nieuwegein-nl.translate.goog/?_x_tr_sl=nl&_x_tr_tl=en
- Bij small viewport link (label: Translate) in header (positie: inline-end).
- Bij large viewport link (icon: translate + label: 'Translate') in header (positie: inline-end)
- Aparte pagina met uitleg over Google Translate en meegeleverde cookies.
- Google Translate balk valt over header na activeren.

**Comment by jeffreylauwers** [comment]
## Rotterdam

Geen specifiek component gevonden in Figma of Storybook. Wel een mogelijkheid om van taal te wisselen op https://www.rotterdam.nl/

---

### Screenshots / Varianten

**Website**

Small viewport
![Scherm­afbeelding 2024-05-01 om 16 16 48](https://github.com/nl-design-system/backlog/assets/545207/17bda2da-72a5-4efb-ba6c-f85acbb2586d)

Small viewport - Nederlands (menu)
![Scherm­afbeelding 2024-05-01 om 16 16 55](https://github.com/nl-design-system/backlog/assets/545207/ec41767b-b0cc-410b-8395-2774f3623412)

Small viewport - Nederlands (menu - translate active)
![Scherm­afbeelding 2024-05-01 om 16 17 06](https://github.com/nl-design-system/backlog/assets/545207/8590cc53-e740-4265-86e5-03cf6abed467)

Small viewport - Engels
![Scherm­afbeelding 2024-05-01 om 16 17 34](https://github.com/nl-design-system/backlog/assets/545207/ac7ad674-428b-4f09-ac9a-d5534e0f0dca)

Small viewport - Engels (menu)
![Scherm­afbeelding 2024-05-01 om 16 17 48](https://github.com/nl-design-system/backlog/assets/545207/1d0d359c-bcc5-4d54-9a14-68a757f9f1f0)

Large viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 16 18 26](https://github.com/nl-design-system/backlog/assets/545207/3cfe414b-3c6c-45eb-bec4-c87514142846)

Large viewport - Nederlands (translate active)
![Scherm­afbeelding 2024-05-01 om 16 18 35](https://github.com/nl-design-system/backlog/assets/545207/0050f865-bc55-4f77-a8b4-7cf293b6885b)

Large viewport - Engels
![Scherm­afbeelding 2024-05-01 om 16 18 58](https://github.com/nl-design-system/backlog/assets/545207/124c8465-0433-4cbc-967f-cd9f0fc7cf72)

---

### Notities

- Keuze tussen link naar Engelse versie en Tourist info of vertalen via Deepl door middel van Select component.
- URL wijzigt bij keuze voor engels naar: https://www.rotterdam.nl/en (en opent nieuwe tab!)
- Bij small viewport als laatste item van 'hamburger' menu.
- Bij large viewport als link (label: 'Translate' icon: globe) in header (positie: inline-end).
- Label blijft 'Translate'.

**Comment by jeffreylauwers** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/components/language-selector/)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=2874-18670&mode=design&t=L30uq8SPFodQQSQF-4)
- Mogelijkheid om van taal te wisselen op https://www.rvo.nl/ (footer)

---

### Screenshots / Varianten

**Figma**

![Scherm­afbeelding 2024-05-01 om 16 31 17](https://github.com/nl-design-system/backlog/assets/545207/c0367653-0e24-4ef9-bb50-f9313beaf561)

**Website**

Small viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 16 32 14](https://github.com/nl-design-system/backlog/assets/545207/4c90db32-efde-4e03-9797-8fab91bf8b05)

Small viewport - Engels
![Scherm­afbeelding 2024-05-01 om 16 33 06](https://github.com/nl-design-system/backlog/assets/545207/72c8ec7e-e036-446b-8d93-3f2de154c856)

Large viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 16 32 02](https://github.com/nl-design-system/backlog/assets/545207/4fa4d0b3-2262-4d28-ba72-2681de27f1ea)

---

### Notities

Documentatie website en Figma
- Documentatie beschikbaar
- Icon: globe + Label: 'English'
- Label wisselt van 'English' naar 'Nederlands'.

Website
- Link in footer
- Twee talen: Engels en Nederlands
- URL wijzigt bij keuze naar: https://english.rvo.nl/

**Comment by jeffreylauwers** [comment]
## Utrecht

Geen specifiek component gevonden in Figma of Storybook. Wel een mogelijkheid om van taal te wisselen op https://www.utrecht.nl/

---

### Screenshots / Varianten

**Website**

Small viewport
![Scherm­afbeelding 2024-05-01 om 16 37 40](https://github.com/nl-design-system/backlog/assets/545207/dc957001-69c8-4dce-9dc6-c52580bf8d26)

Small viewport - Nederlands (menu)
![Scherm­afbeelding 2024-05-01 om 16 37 46](https://github.com/nl-design-system/backlog/assets/545207/753a5333-7606-4987-afb0-26a4d939e1da)

Small viewport - Engels
![Scherm­afbeelding 2024-05-01 om 16 38 01](https://github.com/nl-design-system/backlog/assets/545207/ebe8f40c-7e32-41e8-b8b6-09aa2307b730)

Large viewport - Nederlands
![Scherm­afbeelding 2024-05-01 om 16 38 32](https://github.com/nl-design-system/backlog/assets/545207/8acccec0-305d-4c63-a551-0bde167b583c)

Large viewport - Engels
![Scherm­afbeelding 2024-05-01 om 16 38 48](https://github.com/nl-design-system/backlog/assets/545207/38a72b9b-3ebc-4a3f-862f-b493b694ed0e)

---

### Notities

- Twee talen: Engels en Nederlands.
- URL wijzigt bij keuze voor engels naar: https://www.utrecht.nl/city-of-utrecht.
- Bij small viewport als laatste item van 'hamburger' menu. (2 keer?)
- Bij large viewport als link (icon: globe + label: 'English') in header (positie: inline-end).
- Geen mogelijkheid om terug te wisselen naar Nederlands.
- Label blijft dan ook 'English'.

**Comment by jeffreylauwers** [comment]
## Rijksoverheid

Mogelijkheid om van taal te wisselen op https://www.rijksoverheid.nl/ (footer).

---

### Screenshots / Varianten

**Website**

Small viewport - Nederlands
![Scherm­afbeelding 2024-05-02 om 08 52 06](https://github.com/nl-design-system/backlog/assets/545207/a77ed6b3-bfd4-4080-a99e-fd3a6a203ad7)

Small viewport - Engels
![Scherm­afbeelding 2024-05-02 om 08 52 20](https://github.com/nl-design-system/backlog/assets/545207/13421034-112c-4c7f-88e1-c98cb5785857)

Large viewport - Nederlands
![Scherm­afbeelding 2024-05-02 om 08 51 10](https://github.com/nl-design-system/backlog/assets/545207/2302fd3c-5acf-411d-8d4c-dbc496d17036)

Large viewport - Engels
![Scherm­afbeelding 2024-05-02 om 08 51 23](https://github.com/nl-design-system/backlog/assets/545207/2972bcb6-ad24-47e9-bbe1-51c7761a576e)

---

### Notities

- Onderdeel van de footer.
- 5 talen: Nederlands, Engels, Papiamento, Papiamentu en Nederlandse gebarentaal (linkt naar aparte pagina met video's).
- URL wijzigt bij keuze voor engels naar: https://www.government.nl/

**Comment by jeffreylauwers** [comment]
Codepen voorbeeld opgezet door Rian Rietveld en mij ter voorbereiding op deze vraag bij Design Open Dag van april 2024: https://codepen.io/jeffreylauwers/pen/dyLgKpq

**Comment by Rozerinay** [comment]
@Anton-Olivier leuk dat je deze discussie bent gestart! 

Wat vind je ervan om dit component te hernoemen naar '**Language navigation**'? We hebben besloten om het woord 'navigatie' toe te voegen aan componenten die betrekking hebben op navigatie. Denk bijvoorbeeld aan 'Breadcrumb navigation' en 'Side navigation'.

Met de 'Language navigation' navigeer je tussen verschillende content en in essentie linkt dit component naar een andere url.

**Reply by Anton-Olivier** [reply]
Ik had de naam switcher gebruikt omdat het de naam is die de NN group er aan heeft gegeven ([zie eerder gelinkte artikel](https://www.nngroup.com/articles/language-switching-ecommerce/)). Maar als @jeffreylauwers en jij er allebei naar vragen, kan ik daar niet tegenop 😉

**Comment by JeroenduC** [comment]
Over de plek van de [language switcher](https://www.nngroup.com/articles/language-switching-ecommerce/):

> Tip 2: On desktop sites, put the language switcher in the top corners (left or right).
> 
> When automatic detection fails to work or users want to manually change the site language, it’s crucial that the language switcher be easy to find.
> 
> Across time and cultures, we found that users looked to the upper right corner to change a country or language setting. If they couldn’t find it there, they would turn to the top left corner. We observed that in our previous round of US studies and in our most recent study with Chinese users. This behavior makes sense, because [utility features](https://www.nngroup.com/articles/utility-navigation/) in sites are often located in that page area.

**Comment by Anton-Olivier** [comment]
[Figma file - Logius: Language navigation](https://www.figma.com/design/BJ17S28XNFopx4VZ0jFMy2/Taalswitch?node-id=1-1679&t=fJ8ycbu7u3QFxGpH-1)
In dit projectbestand staat verzameld onderzoek met visuele voorbeelden en uittreksels van internet artikelen.

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Language Navigation op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=language+navigation)

**Comment by eslook** [comment]
In een eerdere Heartbeat heeft Sabine de bevindingen gedeeld van het Language Navigation onderzoek:
 
https://www.youtube.com/watch?v=WgA8flJjCZ0

**Comment by FinnWard** [comment]
Vannuit Logius hebben we een implementatie van een language navigation opgezet op basis van het volgende gebruikersonderzoek: https://gebruikersonderzoeken.nl/docs/onderzoek-bekijken/website-gebruik/logius-taalswitcher/

en hier vind je de storybook met implementatie en documentatie terug: https://rijkshuisstijl-community.vercel.app/?path=/docs/rhc-language-navigation--docs
